lib/streamlit/cli.py

```diff
@@ -152,6 +153,11 @@ def main_run(file_or_url, args):
 
 
 def _main_run(file, args=None):
+    # make a copy before we modify it
+    import sys
+
+    command_line = " ".join(sys.argv)
```

Comment:
I think this doesn't work when there are spaces in the arguments. For example:

This: `streamlit run "my file.py"`
will lead to `argv` equal to `["streamlit", "run", "my file.py"]`
which means `command_line` will be `streamlit run my file.py`.

You should instead use:
```python
subprocess.list2cmdline(sys.argv)
```
(this also escapes quotes inside the arguments, if any)

Reply:
Please add tests to cover the cases above.

Reply:
Nice!

Reply:
Here is tricky again because of the `sys.argv`. In a click test, `sys.argv` is the one of the test. I wonder I should re-compose the command line from click and remove the `sys.argv`.

Reply:
I removed `sys.argv` and constructed a command line from `click`

Reply:
And added a test for it.